### PR TITLE
feat(lidar_apollo_segmentation_tvm): port package

### PR DESCRIPTION
**Pull request by ambroise-arm**
## Description

merge https://github.com/autowarefoundation/autoware.universe/pull/893 first

Port apollo_lidar_segmentation from Autoware.Auto, which uses TVM for the inference. Renamed to match the name of the same package that uses TensorRT.

## Related links

Example usage of https://github.com/autowarefoundation/autoware.universe/issues/628

## Tests performed

Ran the unit test locally.

### Building

By default the packages don't get built. It requires neural networks from the `neural_networks_provider` package that only get downloaded with the `DOWNLOAD_ARTIFACTS` cmake variable set. See its [documentation](https://github.com/autowarefoundation/autoware.universe/blob/main/common/neural_networks_provider/design/neural_networks-design.md) for more information.

### Package name

There is already a `lidar_apollo_instance_segmentation` package that uses TensorRT. I've tried to use the same name with a "_tvm" suffix, but it gets too long and breaks the max line size so I've dropped the "instance" in the name. We may want to rename the current `lidar_apollo_instance_segmentation` to `lidar_apollo_segmentation_tensorrt` (or something similar) to align the naming.

### Output

The output messages don't match the ones from `lidar_apollo_instance_segmentation`. It will probably require some follow-up work for them to be interchangeable. But there is no simulator pipeline to test it with at the moment. 

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by angry-crab — CHANGES_REQUESTED** [review]

**Comment by ambroise-arm** [comment]
Thanks for the review! I am working on addressing the comments.

**Comment by ambroise-arm** [comment]
The first force-push is for rebasing, the second force-push is the actual change.
I've tested the new output with the rosbag demo [here](https://autowarefoundation.github.io/autoware-documentation/main/tutorials/ad-hoc-simulation/rosbag-replay-simulation/). The quality of the detection is not quite as good as the TensorRT package, but I believe that can be addressed later.

**Comment by angry-crab** [comment]
Thank you for the update! I tried to run a rosbag test today but my environment was somehow broken and I'm still trying to fix it. Since you were able to run the test, I think this pr is good to go. As for quality of detection, I think we could open up another issue for that.

File: perception/lidar_apollo_segmentation_tvm_nodes/include/lidar_apollo_segmentation_tvm_nodes/lidar_apollo_segmentation_tvm_node.hpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+{
+namespace lidar_apollo_segmentation_tvm_nodes
+{
+/// \brief Object detection node based on neural network inference.
+class LIDAR_APOLLO_SEGMENTATION_TVM_NODES_PUBLIC ApolloLidarSegmentationNode : public rclcpp::Node
+{
+private:
+  const rclcpp::Subscription<sensor_msgs::msg::PointCloud2>::SharedPtr m_cloud_sub_ptr;
+  const rclcpp::Publisher<tier4_perception_msgs::msg::DetectedObjectsWithFeature>::SharedPtr
+    m_box_pub_ptr;
```
Comment (angry-crab) [inline perception/lidar_apollo_segmentation_tvm_nodes/include/lidar_apollo_segmentation_tvm_nodes/lidar_apollo_segmentation_tvm_node.hpp:40]: I think we can rename this variable to something like `m_detected_pub_ptr`.

File: perception/lidar_apollo_segmentation_tvm_nodes/src/lidar_apollo_segmentation_tvm_node.cpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  const sensor_msgs::msg::PointCloud2::SharedPtr & msg)
+{
+  std::shared_ptr<const tier4_perception_msgs::msg::DetectedObjectsWithFeature> output_msg;
+  try {
+    output_msg = m_detector_ptr->detectDynamicObjects(*msg);
+  } catch (const std::exception & e) {
+    RCLCPP_WARN(get_logger(), e.what());
+    return;
+  }
+  m_box_pub_ptr->publish(*output_msg);
```
Comment (angry-crab) [inline perception/lidar_apollo_segmentation_tvm_nodes/src/lidar_apollo_segmentation_tvm_node.cpp:65]: Same here.

**Review by angry-crab — APPROVED** [review]
LGTM

**Comment by ambroise-arm** [comment]
I have created an issue to track the difference in detection quality.
Can you merge the PR? I don't have write access.

**Pull request merged**